src/game/settings.c

```diff
@@ -33,7 +34,14 @@ static int32_t S_ReadUserSettingsATI()
     FileRead(&OptionSoundFXVolume, sizeof(int16_t), 1, fp);
     FileRead(Layout[1], sizeof(int16_t), 13, fp);
     FileRead(&RenderSettings, sizeof(int32_t), 1, fp);
-    FileRead(&GameHiRes, sizeof(int32_t), 1, fp);
+
+    {
+        int32_t resolution_idx;
+        FileRead(&resolution_idx, sizeof(int32_t), 1, fp);
+        CLAMP(resolution_idx, 0, RESOLUTIONS_SIZE - 1);
+        SetGameScreenSizeIdx(resolution_idx);
+    }
```

Comment:
why is this scoped?

Reply:
To reduce the logical scope of `resolution_idx` (it's not an optimization, only for readability not to inflate the function's state should more variables come into play). I plan on using this  idiom more in the future.